### PR TITLE
feat(ctb): Harden `SemverLock` script

### PR DESCRIPTION
**Pull request by clabby**
## Overview

The `SemverLock` script currently hashes the source code of contracts that inherit `Semver`, which catches most
changes to the contract. However, it does not catch changes to the contract's dependencies, which can cause the
init or runtime bytecode to change without us being notified of a required Semver bump. This PR adds the initcode to the construction of the semver lock hash so that these changes are caught.

### CODE REVIEW

**Comment by clabby** [comment]
Current dependencies on/for this PR:
* develop
  * **PR #6545** <a href="https://app.graphite.dev/github/pr/ethereum-optimism/optimism/6545" target="_blank">Graphite</a> 
    * **PR #6585** <a href="https://app.graphite.dev/github/pr/ethereum-optimism/optimism/6585" target="_blank">Graphite</a> 
      * **PR #6586** <a href="https://app.graphite.dev/github/pr/ethereum-optimism/optimism/6586" target="_blank">Graphite</a>  👈

This comment was auto-generated by [Graphite](https://app.graphite.dev/github/pr/ethereum-optimism/optimism/6586?utm_source=stack-comment).

File: packages/contracts-bedrock/scripts/SemverLock.s.sol [hunk 1]
```diff
@@ -27,8 +27,28 @@ contract SemverLock is Script {
...
+            commands[2] = string.concat(
+                "echo \"",
+                _files[i],
+                "\"| sed -E \'s|src/.*/(.+)\\.sol|\\1|\'"
+            );
+            string memory contractName = string(vm.ffi(commands));
+
+            // Parse the artifact to get the contract's initcode hash.
+            bytes memory initCode = vm.getCode(string.concat(
+                "forge-artifacts/",
```
Comment (tynes) [inline packages/contracts-bedrock/scripts/SemverLock.s.sol]: This could be portable if we call out to forge config --json and pull the out field using jq

Comment (clabby) [reply packages/contracts-bedrock/scripts/SemverLock.s.sol]: Oo good idea. Added 🖖

**Review by tynes — APPROVED** [review]
This could be based on develop tbh

**Comment by tynes** [comment]
@clabby Looks like this needs another rebase

**Comment by tynes** [comment]
Thanks for fixing this @clabby - would like to get this in ASAP so i can rebase all of my PRs on top of it

**Review by tynes — APPROVED** [review]

**Comment by clabby** [comment]
Placing mergify ignore on so that we can get #6545 in first.

**Pull request merged**